File: cpp_e9zMZ/cpp/03-vonglap/94.cpp
#include <math.h>

#include <iostream>
using namespace std;

int main() {
    int yourNumber = 100;
    int sum = 0;
    string result = "";

    while (yourNumber > 0) {
        int lastNumber = yourNumber % 10;
        yourNumber /= 10;
        sum += lastNumber;
        string sign = (yourNumber != 0) ? " + " : "";
        result = sign + to_string(lastNumber) + result;
    }
    cout << result << " = " << sum << endl;
}

File: python_ToSXs/python/03-vonglap/94.py
yourNumber = 126

zsum = 0
result = ""

while yourNumber > 0:
    lastNumber = yourNumber % 10
    yourNumber = int(yourNumber / 10)

    zsum += lastNumber
    result = result + str(lastNumber) + " + "

result = result[0:len(result) - 3]
print(f"{result} = {zsum}")
